assignment-2/src/view/SmartComponents/SmartAnswersList.js

```diff
@@ -0,0 +1,55 @@
+import React, {Component} from 'react';
+import questionModel from "../../model/questionModel";
+import AnswersList from "../DumbComponents/AnswersList";
+import sOUserModel from "../../model/sOUserModel";
+import answerModel from "../../model/answerModel";
+import answersListPresenter from "../../presenter/answersListPresenter";
+import sOUserPresenter from "../../presenter/sOUserPresenter";
+
+
+const mapModelStateToComponentState = (questionModel, answerModel, props) => ({
+    question: getQuestionById(questionModel.state.questions, props.match.params.id),
+    newAnswer: answerModel.state.newAnswer
+});
+
+function getQuestionById(questions, id) {
+    for (let i = 0; i < questions.length; i++) {
+        if (questions[i].id === parseInt(id))
```

Comment:
Meh, you are doing the parse int N times. Wouldn't it be nicer to do it only once?